extensions/spring-web/runtime/src/main/java/io/quarkus/spring/web/runtime/ResponseContentTypeResolver.java

```diff
@@ -20,7 +20,9 @@ public static MediaType resolve(HttpHeaders httpHeaders, String... supportedMedi
         Objects.requireNonNull(httpHeaders, "HttpHeaders cannot be null");
         Objects.requireNonNull(supportedMediaTypes, "Supported media types array cannot be null");
 
-        Variant bestVariant = getBestVariant(httpHeaders.getRequestHeader(ACCEPT), getMediaTypeVariants(supportedMediaTypes));
+        List<String> acceptHeaders = httpHeaders.getRequestHeader(ACCEPT);
+        Variant bestVariant = acceptHeaders.isEmpty() ? null
+                : getBestVariant(acceptHeaders, getMediaTypeVariants(supportedMediaTypes));
```

Comment:
This is done because `negotiation.setAcceptHeaders(anEmptyList);` throws a NPE

Reply:
It would be cleaner to do the list check in the `getBesetVariant` method.

Reply:
Indeed